src/components/filters-history/FiltersHistoryItem.tsx

```diff
@@ -0,0 +1,92 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import React from 'react';
+import moment from 'moment';
+import { FiltersHistory } from '../../stores/FiltersHistoryStore';
+import { FilterState } from '../search-panel/SearchPanelFilters';
+import { FiltersState } from './FiltersHistory';
+
+interface Props {
+	item: FiltersHistory;
+	filter: FiltersState;
+}
+
+const FiltersHistoryItem = ({ item, filter }: Props) => {
+	if (!filter) {
+		return null;
+	}
+	function getValuesUpdater<T extends keyof FilterState>(name: T) {
+		return function valuesUpdater<K extends FilterState[T]>(values: K) {
```

Comment:
maybe if we change this to `return function valuesUpdater(values: string | string[])`  we will not need any's bellow